app/uk/gov/hmrc/tradergoodsprofilesdatastore/controllers/GetRecordsController.scala

```diff
@@ -38,34 +37,19 @@ class GetRecordsController @Inject() (
     extends BackendController(cc) {
 
   def getRecords(
-    eori: String,
-    lastUpdatedDate: Option[String],
-    page: Option[Int],
-    size: Option[Int]
+    eori: String
   ): Action[AnyContent] = identify.async { implicit request =>
     implicit val hc: HeaderCarrier = HeaderCarrier().withExtraHeaders("X-Client-ID" -> "TGP-Frontend")
 
-    routerConnector
-      .getRecords(eori, lastUpdatedDate, page, size)(hc)
-      .flatMap {
-        case httpResponse if httpResponse.status == OK =>
-          val recordsResponse = httpResponse.json.as[GetRecordsResponse]
-          recordsRepository
-            .saveRecords(recordsResponse.goodsItemRecords)
-            .map { _ =>
-              Ok(Json.toJson(recordsResponse))
-            }
-            .recover { case ex: Throwable =>
-              InternalServerError(s"Error saving records: ${ex.getMessage}")
-            }
+    val result = for {
+      httpResponse   <- routerConnector.getRecords(eori)(hc)
+      recordsResponse = httpResponse.json.as[GetRecordsResponse]
+      _              <- recordsRepository.saveRecords(recordsResponse.goodsItemRecords)
+    } yield Ok(Json.toJson(recordsResponse))
 
-        case httpResponse =>
-          Future.successful(Status(httpResponse.status)(httpResponse.body))
-      }
-      .recover { case exception =>
-        logger.error(s"Error fetching records: ${exception.getMessage}")
-        InternalServerError(s"Error fetching records: ${exception.getMessage}")
-      }
+    result.recover { case exception =>
```

Comment:
you don't need to save the result, recover and throw, it will automatically do that if you don't save it in a value, so just have:

```
for {
       httpResponse   <- routerConnector.getRecords(eori)(hc)
       recordsResponse = httpResponse.json.as[GetRecordsResponse]
       _              <- recordsRepository.saveRecords(recordsResponse.goodsItemRecords)
     } yield Ok(Json.toJson(recordsResponse))
```